### ISSUE TITLE
USWDS - Bug: File input error not announced to screen reader

### ISSUE THREAD

**Issue by alex-hull** [uswds/uswds#6157]
### Describe the bug

When inputting an incorrect format in the file input, there is no indication to the screen reader that there is an error. While there is text present, this is not announced. This fails criteria [4.1.3, status message](https://www.w3.org/WAI/WCAG21/Understanding/status-messages.html). 

### Steps to reproduce the bug

1. Got to file input component preview
2. Open a screen reader
3. Go to file input for TXT and PDFs
4. Drag and drop an image file
5. Observe the text stating the error, but not error message is announced

### Expected Behavior

The user using the screen reader would hear that they have input the incorrect format.

### System setup

Chrome, PC
Screen reader used: JAWS

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://designsystem.digital.gov/about/community/#community-conduct).
- [x] I checked the [current issues](https://github.com/uswds/uswds/issues?q=is%3Aissue+is%3Aopen+label%3A%22Type%3A+Bug%22) for duplicate bug reports.

**Comment by mlloydbixal** [comment]
I was able to recreate this issue with the steps to reproduce.

Next steps will be to add the correct role to the div containing the error message. Here is an example: 

<div class="usa-file-input__accepted-files-message" **role="alert"**>This is not a valid file type.</div>

**Comment by mahoneycm** [comment]
## Additional findings from testing:

This appears to be a bit of an edge case. Screen reader users are often navigating via keyboard and mouse and will not be dragging and dropping files in this way.

Luckily, when navigating with keyboard and mouse, the operating system will disable selection of non-valid file types, preventing the error from happening in the first place.

There are edge cases where it's possible to select an invalid file type and upload the file, in which case the screen reader does fail to read out the error message unless navigating line by line via keyboard and mouse.

**Comment by mejiaj** [comment]
@mahoneycm have we looked at the solution that Gov UK has for error messages?

They use a paragraph tied to the inputs `aria-describedby` to connect them.

```html
<!-- Gov.uk File Upload Error -->
<p id="file-upload-1-error" class="govuk-error-message">
  <span class="govuk-visually-hidden">Error:</span> The CSV must be smaller than 2MB
</p>
<input class="govuk-file-upload govuk-file-upload--error" id="file-upload-1" name="fileUpload1" type="file" aria-describedby="file-upload-1-error">
```

https://design-system.service.gov.uk/components/file-upload/#error-messages

**Comment by mejiaj** [comment]
@mahoneycm, thanks I wanted to make sure we evaluated all solutions. Your comment shows it would require additional effort to make it work how we'd like, which might be out of scope for the immediate issue.
